schema_salad/tests/test_codegen_errors.py

```diff
@@ -0,0 +1,220 @@
+.*test8\.cwl:10:5:\s+the\s+`scatterMethod`\s+field\s+is\s+not\s+valid\s+because:
+\s+contains\s+undefined\s+reference\s+to
+\s+`file:///.*/tests/test_schema/abc`$"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message9() -> None:
+    t = "test_schema/test9.cwl"
+    match = r"""^.*test9.cwl:2:1:\s+Object\s+`.*test9.cwl`\s+is\s+not\s+valid\s+because
+\s+tried\s+`Workflow`\s+but
+.*test9\.cwl:8:1:\s+the\s+`steps`\s+field\s+is\s+not\s+valid\s+because:
+\s+tried\s+`array<WorkflowStep>`\s+but
+.*test9\.cwl:9:3:\s+checking object\s+`.*test9\.cwl#step1`
+\s+tried\s+`WorkflowStep`\s+but
+\s+\*\s+missing\s+required\s+field\s+`run`
+.*test9\.cwl:10:5:\s+the\s+`scatterMethod`\s+field\s+is\s+not\s+valid\s+because:
+\s+Expected\s+one\s+of\s+\("dotproduct|nested_crossproduct|flat_crossproduct",\s+"dotproduct|nested_crossproduct|flat_crossproduct",\s+"dotproduct|nested_crossproduct|flat_crossproduct"\) was\s+<class 'int'>"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message10() -> None:
+    t = "test_schema/test10.cwl"
+    match = r"""^.*test10\.cwl:2:1:\s+Object\s+`.*test10\.cwl`\s+is\s+not\s+valid\s+because
+\s+tried\s+`Workflow`\s+but
+.*test10\.cwl:8:1:\s+the\s+`steps`\s+field\s+is\s+not\s+valid\s+because:
+\s+tried\s+`array<WorkflowStep>`\s+but
+.*test10\.cwl:9:3:\s+checking\s+object\s+`.*test10.cwl#step1`
+\s+tried\s+`WorkflowStep`\s+but
+\s+\*\s+missing\s+required\s+field\s+`run`
+.*test10\.cwl:10:5:\s+the\s+`scatterMethod`\s+field\s+is\s+not\s+valid\s+because:
+\s+Expected\s+one\s+of\s+\("dotproduct|nested_crossproduct|flat_crossproduct",\s+"dotproduct|nested_crossproduct|flat_crossproduct",\s+"dotproduct|nested_crossproduct|flat_crossproduct"\)\s+was\s+<class 'ruamel\.yaml\.comments\.CommentedSeq'>"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+def test_error_message11() -> None:
+    t = "test_schema/test11.cwl"
+    match = r"""^.*test11\.cwl:2:1:\s+Object\s+`.*test11.cwl`\s+is\s+not\s+valid\s+because
+\s+tried\s+`Workflow`\s+but
+.*test11\.cwl:8:1:\s+the\s+`steps`\s+field\s+is\s+not\s+valid\s+because:
+\s+tried\s+`array<WorkflowStep>`\s+but
+.*test11\.cwl:9:3:\s+checking\s+object\s+`.*test11\.cwl#step1`
+\s+tried\s+`WorkflowStep`\s+but
+.*test11\.cwl:10:5:\s+the\s+`run`\s+field\s+is\s+not\s+valid\s+because:
+\s+contains\s+undefined\s+reference\s+to
+\s+`file:///.*/tests/test_schema/blub\.cwl`$"""
+    path = get_data("tests/" + t)
+    assert path
+    with pytest.raises(ValidationException, match=match):
+        load_document_by_uri(path)
+
+
+# `loadContents`,\s+`position`,\s+`prefix`,\s+`separate`,\s+`itemSeparator`,\s+`valueFrom`,\s+`shellQuote`
+def test_error_message15() -> None:
```

Comment:
```
tests/test_schema/test15.cwl:3:1:   Object `test15.cwl` is not valid because
                                     tried `CommandLineTool` but
tests/test_schema/test15.cwl:6:1:       the `inputs` field is not valid because:
tests/test_schema/test15.cwl:7:3:         checking object `test15.cwl#message`
                                           tried `CommandInputParameter` but
tests/test_schema/test15.cwl:9:5:             the `inputBinding` field is not valid because:
                                               tried `CommandLineBinding` but
                                                 tried `CommandLineBinding` but
tests/test_schema/test15.cwl:11:7:                 * invalid field `invalid_field`, expected one
                                                   of: `loadContents`, `position`, `prefix`, `separate`,
                                                   `itemSeparator`, `valueFrom`, `shellQuote`
tests/test_schema/test15.cwl:12:7:                 * invalid field `another_invalid_field`,
                                                   expected one of: `loadContents`, `position`, `prefix`,
                                                   `separate`, `itemSeparator`, `valueFrom`, `shellQuote`
```


This prints ` tried 'CommandLineBinding'` twice when it probably doesn't need to print it at all.